[DiD] S6 Gameplay and dialogue revision

nemaara opened the pull request:
Gameplay and dialogue revision for DiD S6, I still need to fix the bigmap so that it shows the correct path to match the new dialogue. This one also needs a bit of playtesting; I only did a very quick test on hard mode to ensure that it was playable, but I'll need to go back later to confirm that it's reasonably balanced (some help here would be appreciated). 

This should address issue #3710.

stevecotton commented on data/campaigns/Descent_Into_Darkness/scenarios/06_Return_to_Parthyn.cfg:
```diff
@@ -259,53 +279,181 @@
+                [message]
+                    side=2
+                    canrecruit=yes
+                    message= _ "Grunts! Prepare the camp! We attack the city tonight—wait. You! Necromancer, you followed us all the way here?!"
```
This message is duplicated in both parts of the if statement. However, there's a single-space change between the two copies (which I assume isn't intentional).

nemaara replied:
Thanks for pointing that out, I missed it.

stevecotton commented on data/campaigns/Descent_Into_Darkness/scenarios/06_Return_to_Parthyn.cfg:
```diff
@@ -259,53 +279,181 @@
+
+        [message]
+            speaker=Malin Keshar
+            message= _ "Drogan! With my newfound powers, I have already eliminated two orc chieftains! If you will allow me to help, I will crush the one who comes here as well!"
```
I count four orcish warlords on eliminated on screen (2 in S4, 2 in S5), plus the off-screen ones mentioned in S5's intro.

nemaara replied:
I think it refers to the ones only in S5, at least that's how I read it in the original. I could always change it to "many", though.

nemaara commented:
Now added dialogue when dark adepts or their advancements get placed. Also changed the wording from 'two orcish chieftains' to 'several'. 

Killing Dela now lets you beat the scenario, but it's not listed as an objective since the player really shouldn't be doing that.

stevecotton commented on data/campaigns/Descent_Into_Darkness/scenarios/06_Return_to_Parthyn.cfg:
```diff
@@ -259,53 +279,242 @@
+
+                [message]
+                    speaker=unit
+                    message= _ "I joined Malin of my own free will! Drogan, you are a fool who would let the orcs slay us all simply because you do not understand the power that necromancy holds."
```
Thanks for adding this, it sounds good.

It should probably have a female_message too.

nemaara replied:
Added, did I do it properly?

stevecotton replied:
The (AFAIK undocumented) coding convention seems to be to use `message=` for the male message.

stevecotton replied:
Sorry to keep adding comments, but please consider whether it should have an [allow_undo][/allow_undo], like the hint in #3422 (which would allow undoing a recall, but not a recruit).

I'm not sure if it should or shouldn't have this - strategy-wise, it seems better to allow it; but story-wise it's another irrevocable small step on Malin's descent.

nemaara replied:
I don't think I should allow undoing because I feel like it would look really strange if the text fires but then you just undo recalling your unit.

stevecotton commented on data/campaigns/Descent_Into_Darkness/scenarios/06_Return_to_Parthyn.cfg:
```diff
@@ -259,53 +279,242 @@
+
+                [message]
+                    speaker=Dela Keshar
+                    message= _ "No quip about how we do not understand you? No comment about your moral superiority? Nothing to say at all? I can ony hope you are so quiet when I kill you."
```
Spelling error: "ony"

nemaara commented on data/campaigns/Descent_Into_Darkness/scenarios/06_Return_to_Parthyn.cfg:
```diff
@@ -330,22 +534,47 @@
+
+                [message]
+                    speaker=Drogan
+                    message= _ "At least you have not completely lost your humanity by allying yourself with the orcs. But you cannot return to Parthyn, Malin. No necromancer will ever be welcome here."
```
Discussed in DM, I'll put reasoning here for others to see. 

I don't think most characters do speak (or need to speak) with perfect grammar, and at least in modern speech, it seems like a lot of people do use "but" to begin a sentence (yes, after a full stop). Speaking this aloud reads just fine to me (an alternative is to use ellipses before the 'but', but people have been complaining about having too many of them so I'm hesitant to change it to that).

nemaara commented on data/campaigns/Descent_Into_Darkness/scenarios/06_Return_to_Parthyn.cfg:
```diff
@@ -4,24 +4,28 @@
+            {STORYTXT_BACKGROUND parthyn.jpg}
+        [/part]
+        [part]
+            story= _ "Malin tramps through the woods tirelessly, forsaking several days’ sleep as he and his servants draw closer to the Great River, hoping to cut off the orcs before they reach Parthyn. Soon, he recognizes some familiar surroundings..."
```
Implemented.
